src/transformers/models/bart/modeling_bart.py

```diff
@@ -1058,14 +1320,22 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input) * self.embed_scale
 
+        if attention_mask is None:
+            padding_mask = None
+        else:
+            padding_mask = attention_mask if 0 in attention_mask else None
```

Comment:
In case we decide to cache the expanded mask, please make sure to keep the logic --> if no padding tokens then attention_mask = None to not enter in the `if padding_mask is not None:` condition that will pad / unpad the input